src/domain/enrolment/__tests__/CreateEnrolmentPage.test.tsx

```diff
@@ -398,66 +398,94 @@ test('render and focuses error notification correctly', async () => {
-  userEvent.type(screen.getByLabelText(/aikuisia/i), '2');
-  userEvent.tab();
-  await waitFor(() => {
-    expect(
-      screen.getByText(
-        /Arvon tulee olla enintään 1 yhdessä lasten lukumäärän kanssa/i
-      )
-    ).toBeInTheDocument();
+describe('max group size validation of the Children and Adults -fields', () => {
+  test('both of the fields are greater than the max group size', async () => {
+    render(<CreateEnrolmentPage />, {
+      mocks: mock3,
+      query: { eventId: eventId, occurrences: occurrenceIds },
+    });
+    await waitFor(() => {
+      expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
+    });
+    userEvent.type(screen.getByLabelText(/lapsia/i), '21');
+    userEvent.type(screen.getByLabelText(/aikuisia/i), '22');
+    userEvent.tab();
+    await waitFor(() => {
+      expect(
+        screen.getAllByText(
+          /Lasten ja aikuisten yhteislukumäärän tulee olla enintään 20/i
+        )
+      ).toHaveLength(2);
+    });
   });
-});
 
-test('amount of adults is limited to 0 when children fills the group in whole', async () => {
-  render(<CreateEnrolmentPage />, {
-    mocks: mock3,
-    query: { eventId: eventId, occurrences: occurrenceIds },
-  });
-  await waitFor(() => {
-    expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
+  test('one field is greater than the max group size and another one is (still) empty', async () => {
+    render(<CreateEnrolmentPage />, {
+      mocks: mock3,
+      query: { eventId: eventId, occurrences: occurrenceIds },
+    });
+    await waitFor(() => {
+      expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
+    });
+    userEvent.type(screen.getByLabelText(/lapsia/i), '21');
+    userEvent.type(screen.getByLabelText(/aikuisia/i), '');
+    userEvent.tab();
+    await waitFor(() => {
+      expect(
+        screen.getByText(
+          /Lasten ja aikuisten yhteislukumäärän tulee olla enintään 20/i
+        )
+      ).toBeInTheDocument();
+    });
+    expect(screen.getByText(/Tämä kenttä on pakollinen/i)).toBeInTheDocument();
   });
-  userEvent.type(screen.getByLabelText(/lapsia/i), '20');
-  userEvent.type(screen.getByLabelText(/aikuisia/i), '2');
-  userEvent.tab();
-  await waitFor(() => {
+
+  test('both the fields are valid as a single, but the total is greater than the maximum group size', async () => {
+    render(<CreateEnrolmentPage />, {
+      mocks: mock3,
+      query: { eventId: eventId, occurrences: occurrenceIds },
+    });
+    await waitFor(() => {
+      expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
+    });
+    userEvent.type(screen.getByLabelText(/lapsia/i), '19');
+    userEvent.type(screen.getByLabelText(/aikuisia/i), '18');
+    userEvent.tab();
+    await waitFor(() => {
+      expect(
+        screen.getByText(
+          /Arvon tulee olla enintään 2 yhdessä aikuisten lukumäärän kanssa/i
+        )
+      ).toBeInTheDocument();
+    });
```

Comment:
Some common helper test funciton could be constructed to avoid this repetition. Maybe pass those error messages and amount of kids and adults as parameter

Reply:
I'm now creating the form in a common function: 

```
    createEnrolmentForm('19', '18');
    await waitFor(() => {
      expect(
        screen.getByText(
          /Arvon tulee olla enintään 2 yhdessä aikuisten lukumäärän kanssa/i
        )
      ).toBeInTheDocument();
    });
```